core/blockchain_test.go

```diff
@@ -688,7 +683,7 @@ func TestBlocksInsertNonceError(t *testing.T)  { testInsertNonceError(t, true) }
 func testInsertNonceError(t *testing.T, full bool) {
 	for i := 1; i < 25 && !t.Failed(); i++ {
 		// Create a pristine chain and database
-		db, blockchain, err := newCanonical(ethash.NewFaker(), 0, full)
+		gendb, _, blockchain, err := newCanonical(ethash.NewFaker(), 0, full)
```

Comment:
I'm not overly fond of `gendb`. Should be either
a) `db` like before
b) `genesisDb`, spelled out, or  
c) `genDb` since it's an abbreviation

Reply:
a) I won't vote db, since we deliberately to distinguish the database for blockchain and the database for chain maker, it's super easy to mess them up
b) `genesisDb` more fells like it's s database contains genesis state, but it's not the case here, it can contain states of generated block
c) `genDb` looks good to me, should we go with this direction?

Reply:
used `genDb`